packages/contractkit/src/wallets/signers/azure-key-vault-client.ts

```diff
@@ -0,0 +1,181 @@
+// @ts-ignore-next-line
+import { account as Account, nat as Nat } from 'eth-lib'
+import { DefaultAzureCredential } from '@azure/identity'
+import { KeyClient, CryptographyClient, KeyVaultKey } from '@azure/keyvault-keys'
+// @ts-ignore-next-line
+import BN, { BN } from 'bn.js'
+import { ec as EC } from 'elliptic'
+import { ecdsaRecover } from 'secp256k1'
+
+/**
+ * Provides an abstraction on Azure Key Vault for performing signing operations
+ */
+export class AzureKeyVaultClient {
+  private readonly vaultName: string
+  // Unique URI of the Azure Key Vault
+  private readonly vaultUri: string
+  private readonly credential: DefaultAzureCredential
+  private readonly keyClient: KeyClient
+  private readonly SIGNING_ALGORITHM: string = 'ECDSA256'
+
+  // 0x04 prefix indicates that the key is not compressed
+  // https://tools.ietf.org/html/rfc5480#section-2.2
+  private readonly publicKeyPrefix: number = 0x04
+
+  constructor(vaultName: string) {
+    this.vaultName = vaultName
+    this.vaultUri = `https://${this.vaultName}.vault.azure.net`
+    // DefaultAzureCredential supports service principal or managed identity
+    // If using a service principal, you must set the appropriate environment vars
+    this.credential = new DefaultAzureCredential()
```

Comment:
Could you explain how we expect someone using this to provide credentials?
I think it'd be nice to not rely on env vars in here and instead pass in specific config secrets to contractkit, but if that takes a big refactor I'd like to understand more what's expected in the env vars

Reply:
The env vars needed are outlined in the .env template. I'll add a write-up in our dev readme on where to get these from. I just added a mock AzureKeyVaultClient for unit testing. This is convenient for devs not touching the HSM logic, but shouldn't be solely relied upon if modifying the signing logic.